### PR TITLE
Fix conda check further

### PR DESCRIPTION
**Pull request by brisvag**
# Description
On my machine #5496 was not enough to fix the issue, and I was still getting this error:

```py
_______________________________________________________________________ ERROR collecting napari/_qt/dialogs/_tests/test_installer_process.py _______________________________________________________________________
napari/_qt/dialogs/_tests/test_installer_process.py:185: in <module>
    not CondaInstallerTool.available(), reason="Conda is not available."
napari/_qt/dialogs/qt_package_installer.py:176: in available
    return call([executable, "--version"]) == 0
/usr/lib/python3.10/subprocess.py:345: in call
    with Popen(*popenargs, **kwargs) as p:
/usr/lib/python3.10/subprocess.py:971: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
/usr/lib/python3.10/subprocess.py:1847: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   FileNotFoundError: [Errno 2] No such file or directory: 'conda'

```

This PR moves the whole check inside the try block to catch this as well.

cc @czaki

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/_qt/dialogs/qt_package_installer.py [hunk 1]
```diff
@@ -171,9 +171,9 @@ def executable(cls):
     def available(cls):
         try:
             executable = cls.executable()
```
Comment (Czaki) [inline napari/_qt/dialogs/qt_package_installer.py]:
```suggestion
        executable = cls.executable()
        try:
```

**Review by Czaki — APPROVED** [review]
I made a change to exclude command that cannot raise this exception outside the try block. 

Fell free to revert it and merge which version you prefer. 
I think that this pr does not require 24-h waitings.

**Comment by brisvag** [comment]
agreed! I guess we should just merge despite the failures due to leaked widgets, as they are unrelated.

**Comment by brisvag** [comment]
There was also also a typechekcing failure due to unused `#type: ignore` comment. I saw it pop up elsewhere as well. I removed it here, should be good?

**Comment by Czaki** [comment]
Yes. I already have removed it in pre-commit pr

**Pull request closed without merging**